src/hazelcore/Statics_Exp.re

```diff
@@ -1204,18 +1221,39 @@ and ana_fix_holes_operand =
       };
     }
   | Case(_, scrut, rules) =>
-    let (scrut, scrut_ty, u_gen) =
-      syn_fix_holes(ctx, u_gen, ~renumber_empty_holes, scrut);
-    let (rules, u_gen) =
-      ana_fix_holes_rules(
-        ctx,
-        u_gen,
-        ~renumber_empty_holes,
-        rules,
-        scrut_ty,
-        ty,
-      );
-    (Case(StandardErrStatus(NotInHole), scrut, rules), u_gen);
+    switch (ty) {
+    | Unknown(SynPatternVar) =>
+      let (op, _, u_gen) =
+        syn_fix_holes_operand(ctx, u_gen, ~renumber_empty_holes, e);
+      (op, u_gen);
+    | Unknown(Internal | TypHole) =>
```

Comment:
I don't think these cases need to be special cased here, they should go through the catchall case -- need to mark the situation where there are inconsistent branches even when analyzing against a non-hole type